apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/JobsTest.cs

```diff
@@ -67,5 +69,210 @@ public void ListJobs_FullProjection()
+            var originTable = bqClient.GetTable(_fixture.DatasetId, _fixture.HighScoreTableId);
+            var destinationBucket = _fixture.StorageBucketName;
+            var destinationObject = _fixture.GenerateStorageObjectName();
+            var destinationUri = $"gs://{destinationBucket}/{destinationObject}";
+            var destinationReference = bqClient.GetTableReference(_fixture.DatasetId, _fixture.CreateTableId());
+
+            // Just extracting the data into GCS
+            var extractJob = originTable.CreateExtractJob(destinationUri);
+            extractJob = extractJob.PollUntilCompleted().ThrowOnAnyError();
+
+            var loadJob = bqClient.CreateLoadJob(
+                destinationUri, destinationReference, originTable.Schema, 
+                new CreateLoadJobOptions() { SkipLeadingRows = 1, Labels = JobLabels });
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+
+            loadJob = loadJob.PollUntilCompleted().ThrowOnAnyError();
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersKey()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"This is Invalid & throws", "this_is_valid-2" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersValue()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"this_is_valid-2", "This is Invalid & throws" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        internal static IDictionary<string, string> JobLabels =>
+            new Dictionary<string, string>()
+            {
+                // This one is a little absurd for jobs because labels can only be set
+                // on job creation. This means not to set the label because this is copying
+                // the bahaviour of labels for other resources like DataSets and Table.
+                // Testing for it anyways because the api allows is.
+                { "one_label", null },
+                { "another-label-2", "label_value_2" },
+                { "yet_another_label", string.Empty }
```

Comment:
My personal preference is for "" over string.Empty, but I don't much mind either way (and I don't think it's worth worrying about :)

Reply:
Changed to "" to try and keep everything the same, I don't have a conscious preference, I'm just used to string.Empty I think.